src/datepicker/datepicker-month-view.spec.ts

```diff
@@ -0,0 +1,151 @@
+  expect(result).toEqual(weekdays);
+}
+
+function expectWeekNumbers(element: HTMLElement, weeknumbers: string[]) {
+  const result = getWeekNumbers(element).map(td => td.innerText.trim());
+  expect(result).toEqual(weeknumbers);
+}
+
+function expectDates(element: HTMLElement, dates: string[]) {
+  const result = getDates(element).map(td => td.innerText.trim());
+  expect(result).toEqual(dates);
+}
+
+describe('ngbDatepickerMonthView', () => {
+
+  beforeEach(() => {
+    TestBed.overrideModule(NgbDatepickerModule, {set: {exports: [NgbDatepickerMonthView, NgbDatepickerDayView]}});
+    TestBed.configureTestingModule({declarations: [TestComponent], imports: [NgbDatepickerModule]});
+  });
+
+  it('should show/hide weekdays', () => {
+    const fixture =
+        createTestComponent('<tbody ngbDatepickerMonthView [month]="month" [showWeekdays]="showWeekdays"></tbody>');
+
+    expectWeekdays(fixture.nativeElement, ['Mo']);
+
+    fixture.componentInstance.showWeekdays = false;
+    fixture.detectChanges();
+    expectWeekdays(fixture.nativeElement, []);
+  });
+
+  it('should show/hide week numbers', () => {
+    const fixture = createTestComponent(
+        '<tbody ngbDatepickerMonthView [month]="month" [showWeekNumbers]="showWeekNumbers"></tbody>');
+
+    expectWeekNumbers(fixture.nativeElement, ['2']);
+
+    fixture.componentInstance.showWeekNumbers = false;
+    fixture.detectChanges();
+    expectWeekNumbers(fixture.nativeElement, []);
+  });
+
+  it('should use custom template to display dates', () => {
+    const fixture = createTestComponent(`
+        <template #tpl let-date="date">{{ date.day }}</template>
+        <tbody ngbDatepickerMonthView [month]="month" [dayTemplate]="tpl"></tbody>
+      `);
+    expectDates(fixture.nativeElement, ['22']);
+  });
+
+  it('should send date selection events', () => {
+    const fixture = createTestComponent(`
+        <template #tpl let-date="date">{{ date.day }}</template>
+        <tbody ngbDatepickerMonthView [month]="month" [dayTemplate]="tpl" (select)="onClick($event)"></tbody>
+      `);
+
+    spyOn(fixture.componentInstance, 'onClick');
+
+    const dates = getDates(fixture.nativeElement);
+    dates[0].click();
+
+    expect(fixture.componentInstance.onClick).toHaveBeenCalledWith(new NgbDate(2016, 7, 22));
+  });
+
+  it('should not send date selection events for disabled dates', () => {
+    const fixture = createTestComponent(`
+        <template #tpl let-date="date">{{ date.day }}</template>
+        <tbody ngbDatepickerMonthView [month]="month" [dayTemplate]="tpl" (select)="onClick($event)"></tbody>
+      `);
+
+    fixture.componentInstance.month.weeks[0].days[0].disabled = true;
+    fixture.detectChanges();
+
+    spyOn(fixture.componentInstance, 'onClick');
+
+    const dates = getDates(fixture.nativeElement);
+    dates[0].click();
+
+    expect(fixture.componentInstance.onClick).not.toHaveBeenCalledWith();
+  });
+
+  // CURSOR TESTS FAIL IN IE 9
```

Comment:
We are going to add a test utility for browser sniffing and skip those tests in IE9